### PR TITLE
Adds Gauges to metrics

### PR DESCRIPTION
**Pull request by topicus**
- Add Gauges and their corresponding tests

### CODE REVIEW

**Comment by aryszka** [comment]
:+1:

**Comment by aryszka** [comment]
Is it possible to apply these changes for the prometheus variant as well? If yes, then it would also be possible to declare these methods on `interface Metrics`, and they would become available in the filters as custom metrics. It's also for consistency.

**Comment by topicus** [comment]
I tried to make it for both but prometheus doesn't provide Meters. I can remove Meters from this PR and implement the prometheus version of Gauges.

I think this can provide a good balance between adding value and keeping consistency.

**Comment by aryszka** [comment]
It would be really nice to have them, or at least the Gauges on the shared Metrics interface. Do you think it is better to do it as part of this PR, or is it more reasonable to do it as a follow-up?

**Comment by topicus** [comment]
I have updated the PR. It includes the Gauges implementation for Prometheus and CodeHale with their respective tests.

Later we can figure out how to implement Meters in a way that works for both clients.

File: metrics/codahale.go [hunk 1]
```diff
@@ -282,6 +295,13 @@ func (sm skipperMetrics) MarshalJSON() ([]byte, error) {
 		case metrics.Gauge:
 			metricsFamily = "gauges"
 			values["value"] = m.Value()
+		case metrics.Meter:
```
Comment (szuecs) [inline metrics/codahale.go]: this should be dropped for this PR, right?

**Comment by topicus** [comment]
@szuecs I fixed those items. Could you review again?

**Comment by aryszka** [comment]
:+1:

**Comment by topicus** [comment]
:+1:

**Comment by szuecs** [comment]
@topicus can you also add some documentation in https://github.com/zalando/skipper/blob/master/docs/operations.md#monitoring please

**Comment by topicus** [comment]
:+1:

**Comment by szuecs** [comment]
@topicus please do https://github.com/zalando/skipper/pull/726#issuecomment-408409361, before merge, thanks!

**Comment by topicus** [comment]
@aryszka  @szuecs Is there any section like "extending skipper" or similar? All the documentation I found is about using skipper out-of-the-box except by the plugins section.

**Comment by szuecs** [comment]
@topicus I am not sure what you mean by that.
I you like to chat about that, please join our hipchat channel, I guess it'sbetter to discuss it and document if there is a need to.

**Comment by szuecs** [comment]
As discussed offline there are no additional metrics, only a facilty to add Gauges if skipper is used as library.

**Comment by szuecs** [comment]
:+1:

**Pull request merged**